apps/src/lib/node/ledger/events/log/dumb_queries.rs

```diff
@@ -0,0 +1,156 @@
+//! Silly simple Tendermint query parser.
+//!
+//! This parser will only work with simple queries of the form:
+//!
+//! ```text
+//! tm.event='NewBlock' AND <accepted|applied>.<$attr>='<$value>'
+//! ```
+
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use crate::node::ledger::events::{Event, EventType};
+
+/// Regular expression used to parse Tendermint queries.
+const QUERY_PARSING_REGEX_STR: &str =
+    r"^tm\.event='NewBlock' AND (accepted|applied)\.([\w_]+)='([^']+)'$";
+
+lazy_static! {
+    /// Compiled regular expression used to parse Tendermint queries.
+    static ref QUERY_PARSING_REGEX: Regex = Regex::new(QUERY_PARSING_REGEX_STR).unwrap();
+}
+
+/// A [`QueryMatcher`] verifies if a Namada event matches a
+/// given Tendermint query.
+#[derive(Debug, Clone)]
+pub struct QueryMatcher<'q> {
+    event_type: EventType,
+    attr: String,
+    value: &'q str,
+}
+
+impl<'q> QueryMatcher<'q> {
+    /// Checks if this [`QueryMatcher`] validates the
+    /// given [`Event`].
+    pub fn matches(&self, event: &Event) -> bool {
+        event.event_type == self.event_type
+            && event
+                .attributes
+                .get(&self.attr)
+                .as_ref()
+                .map(|value| value == &self.value)
+                .unwrap_or_default()
+    }
+
+    /// Parses a Tendermint-like events query.
+    pub fn parse(query: &'q str) -> Option<Self> {
+        let captures = QUERY_PARSING_REGEX.captures(query)?;
+
+        let event_type = match captures.get(1)?.as_str() {
+            "accepted" => EventType::Accepted,
+            "applied" => EventType::Applied,
+            // NOTE: the regex only matches `accepted`
+            // and `applied`
+            _ => unreachable!(),
```

Comment:
The `?` on the regex does that.

Reply:
Oh, I guess it's an option, not a result

Reply:
But, the unreachable!() is still necessary I think.

Reply:
Oh yeah you're right, the capture can only get `"accepted"` or `"applied"` otherwise we won't reach this point, so the `unreachable!()` makes sense. We should return a `Result` still indicating this is what happened (where we are currently returning `None`)

Reply:
we're now logging invalid queries